src/Test/Builder/AbstractStep.php

```diff
@@ -24,4 +24,9 @@ final public function setRuleName(string $ruleName): void
     {
         $this->rule->ruleName = $ruleName;
     }
+
+    final public function addTip(string $tip): void
+    {
+        $this->rule->tips[] = $tip;
+    }
```

Comment:
to keep consistency on the DSL it would be better if this is `because(string ...$tips)`, and it should be a separated builder step that extends BuildStep

Reply:
just to get this right, to be in line with the DSL, we would create a step like this:

```php
class ReasoningOrBuildStep extends BuildStep {
    public function because(string ... $reasons): BuildStep {
        array_map($this->rule->addTips(...), $reasons);
        return new BuildStep($this->rule);
    }
}
```

and change every other return new `BuildStep` to `ReasoningOrBuildStep` ? i am a bit confused, how to deal with for example `TargetExcludeOrBuildStep` since that would be a `TargetExcludeOrReasoningOrBuildStep`? :thinking:

whouldn't it be easier to add this method to the BuildStep class?

i am unsure how to implement this.

Reply:
```php
class TargetExcludeOrBuildStep extends TipOrBuildStep
{
    public function excluding(SelectorInterface ...$selectors): TipOrBuildStep
    {
        $this->rule->targetExcludes = array_values($selectors);

        return new TipOrBuildStep($this->rule);
    }
}
```


```php
class TipOrBuildStep extends BuildStep
{
    public function because(string ...$tips): BuildStep
    {
        $this->rule->tips = array_values($tips);

        return new BuildStep($this->rule);
    }
}
```

Does it make sense to you?

Reply:
a i see, was a bit confused at first.
i added this variant and added tests